test/mpi/buffer.cc

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+#include <iostream>
+#include <mpi.h>
+#include <ucc/api/ucc.h>
+BEGIN_C_DECLS
+#include "core/ucc_mc.h"
+#include "utils/ucc_math.h"
+END_C_DECLS
+
+template<typename T>
+void init_buffer_host(void *buf, size_t count, int _value)
```

Comment:
T *buf instead of void *buf

Reply:
then i will have to do the type case every time i call this init_buffer function because in the test class the buffer is void*.
Right now i just call init_buffer_host(ptr, ...), if wi declare buf argument as typed will i need to call it : init_buffer_host((float*), ...) ?or typecast from "void*" to T *buf will be made by compiler w/o warnings?

Reply:
I think it will cast it implicitly

Reply:
i checked - it won't. even gcc 4.8.5 complains on type mismatch.